main.tf

```diff
@@ -36,7 +36,7 @@ locals {
   api_s_account_fmt      = "${format("serviceAccount:%s", local.api_s_account)}"
   gke_shared_vpc_enabled = "${var.shared_vpc != "" && contains(var.activate_apis, "container.googleapis.com") ? "true" : "false"}"
   gke_s_account          = "${format("service-%s@container-engine-robot.iam.gserviceaccount.com", local.project_number)}"
-  gke_s_account_fmt      = "${format("serviceAccount:%s", local.gke_s_account)}"
+  gke_s_account_fmt      = "${local.gke_shared_vpc_enabled ? format("serviceAccount:%s", local.gke_s_account) : ""}"
```

Comment:
had to add the conditional for the gke service account fmt on line 39 of main.tf as conditionals for lists in the iam bindings doesn't work in .11 (https://github.com/hashicorp/terraform/issues/12453)